small-talk/app/chat/page.js

```diff
@@ -1,14 +1,66 @@
 "use client"
 import Link from 'next/link'
-import React from 'react';
 import ThemeLayout from '../components/ThemeLayout';
+import socket from "../../util/socket";
+import React, { useEffect, useState } from 'react';
 
-export default function chat() {
-    return (
+export default function Chat(){
+	const [text, setText] = useState(null);
+	const [user, setUser] = useState(null);
+	  
+	useEffect(() => {
+		socket.auth = "random";
+    	socket.connect();
+		setText("whatever");
+		
+		// getting all currently connected users
+		socket.on("users", (receivedUsers) => {
+            console.log("Received users:", receivedUsers);
+            receivedUsers.forEach((user) => {
+                user.self = user.userID === socket.id;
+            });
+            setUser(receivedUsers);
+        });
+		
+		// set the event to connected users
+		socket.on("user connected", (user) => {
+			console.log("User connected:", user);
+			this.users.push(user);
+		});
+		
+		// handles connection error
+		socket.on("connect_error", (err) => {
+			if (err.message === "invalid username") {
+				this.usernameAlreadySelected = false;
+			}
+		});
+		
+		// disconnect socket
+		return () => {
+			socket.off("users");
+			socket.off("user connected");
+			socket.off("connect_error");
+		};
+	}, [text]);
```

Comment:
I'm a little confused what the purpose of using text as a dependency in the useEffect, since the only time it's being changed is within the useEffect itself, which I think would cause this useEffect to run twice. Once when the page loads, then again when the text is changed within the useEffect. 

Is this connection intended to run at the button press?